exporter/kafkaexporter/README.md

```diff
@@ -36,6 +36,9 @@ The following settings can be optionally configured:
   - The following encodings are valid *only* for **logs**.
     - `raw`: if the log record body is a byte array, it is sent as is. Otherwise, it is serialized to JSON. Resource and record attributes are discarded.
 - `partition_traces_by_id` (default = false): configures the exporter to include the trace ID as the message key in trace messages sent to kafka. *Please note:* this setting does not have any effect on Jaeger encoding exporters since Jaeger exporters include trace ID as the message key by default.
+- `partition_metrics_by_resource_attributes`  configures the exporter to include the hash of sorted resource attributes as the message partitioning key in metric messages sent to kafka.
+  - `enabled`: (default = false) 
+  - `attributes`: (default = []) the list of resource attributes to include in the partitioning key. If empty, all resource attributes are included.
```

Comment:
Can you please clarify why you need to specify the list of attributes? Why does taking a hash of all the resource attributes doesn't work for you?

Reply:
Although indeed in my use case hashing of all resource attributes would work, It was added by the recommendation from earlier in this PR to add further flexibility for the users.

Reply:
I see the comment from @crobert-1, but I don't understand what use case we would cover by providing this option. I would prefer not to complicate the user interface unless there is a potential need for this. @crobert-1, please clarify what I'm missing.

Thinking of how this can evolve going forward, I can imagine needing to partition by metric names. For example, for further aggregation by datapoint attributes. In that case, we would need to add `partition_metrics_by_metric_name`, which is fine, I guess...

Reply:
One of the use cases I see is following:

Let's say one has several applications in the pod, but for some reason one wants to have all metrics from a single pod to end up in the same topic, so they would specifiy only `service.instance.id` in the attribute list.

Reply:
@dmitryax @crobert-1 I reverted specific resource attribute keys selection for hashing, please have a look again.

Thank you!